README.md

```diff
@@ -631,6 +631,22 @@ x-ratelimit-reset: "1377205443"
 
 See the [Faraday README][faraday] for more middleware magic.
 
+### Retrying
+
+If you want to retry requests, use `Octokit::Middleware::Retry`.
+
+It uses [Faraday Retry gem] for Faraday 2.x. Add the gem to your Gemfile
+
+```ruby
+gem 'faraday-retry'
+```
+
+Next, insert `Octokit::Middleware::Retry` before `Octokit::Response::RaiseError`:
+
+```ruby
+Octokit.middleware.insert Octokit::Response::RaiseError, Octokit::Middleware::Retry
```

Comment:
I would think configuring it on the client would be better
```ruby
Octokit::Client.new(auto_retry: true)
```

and then in the connection sawyer options you can insert the retry middleware if @auto_retry
```ruby
conn_opts[:builder].insert(0, *retry_middleware) if @auto_retry
```

Reply:
While having a parameter for `Octokit::Client.new` is more concise, there are a few advantages of explicitly requiring to interact with middleware:
1. A user can specify additional options.
2. It is clear that it interacts with middleware.
3. It is consistent with the suggestion for caching, which follows this section in `README.md`.

So there is some trade-off. What do you prefer?

Reply:
What happens if you want one client that retries and another that doesn't. Setting on Octokit.middleware is global

Reply:
You can also configure an individual instance; see: https://github.com/octokit/octokit.rb/pull/1706/files#diff-7ee18df1929eee9ac41b3ffa5f60cc4904aeb944ff7be6034721b1a76dc2798dR520